### DISCUSSION TITLE
Github Copilot VS Code extension is not working with NixOS install of VS Code 1.71 - it is not recognised as a recent enough version for Github Copilot

### DISCUSSION THREAD

**Discussion by stuzenz** [community/community#33392] [Copilot Conversations]
**Description**: Visual Studio Code 1.71 on NixOS is not recognised as up-to-date enough for Github Copilot to work
**Steps to reproduce**: NixOS installed version of vs code (22.05 with 1.69 version of vs code and NixOS unstable branch (22.11) with 1.71 vs code version both have this issue)
**Expected behavior**: vs code greater than 1.57 should be compatible for Github Copilot

```bash
code --version
1.71.0
784b0177c56c607789f9638da7b6bf3230d47a8c
x64

nix --version   
nix (Nix) 2.11.0

```

As a side note, on my old distribution (Archlinux) the incompatibility error message did not present itself.

![Screenshot from 2022-09-18 15-18-44](https://user-images.githubusercontent.com/4701809/190884208-f29382be-a15c-4d6e-8213-551f54eb3f84.png)

Thanks!
Stuart

**Comment by stuzenz** [comment]
As a temporary (and seemingly workable) solution, I have succeeded in getting the VS Code GitHub Copilot extension to work using a flatpak install of vs code.

![Screenshot from 2022-09-18 15-44-47](https://user-images.githubusercontent.com/4701809/190884797-d5e4a6a3-428d-4867-a1cb-c855f1c151fe.png)

**Comment by smasher164** [comment]
The copilot vscode extension on nixpkgs is outdated. Add the following to your nix config, where you specify extensions:
```Nix
++ pkgs.vscode-utils.extensionsFromVscodeMarketplace [
  {
    name = "copilot";
    publisher = "GitHub";
    version = "1.46.6822";
    sha256 = "sha256-L71mC0190ZubqNVliu7es4SDsBTGVokePpcNupABI8Q=";
  }
];
```
